subgraphs/seaport/src/mapping.ts

```diff
@@ -0,0 +1,691 @@
+import { log, ethereum, BigInt, Address, Bytes } from "@graphprotocol/graph-ts";
+import {
+  OrderFulfilled,
+  OrderFulfilledConsiderationStruct,
+  OrderFulfilledOfferStruct,
+} from "../generated/SeaportExchange/SeaportExchange";
+import {
+  Collection,
+  CollectionDailySnapshot,
+  Marketplace,
+  MarketplaceDailySnapshot,
+  Trade,
+  _Item,
+} from "../generated/schema";
+import {
+  BIGDECIMAL_HUNDRED,
+  BIGDECIMAL_MAX,
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  ERC1155_INTERFACE_IDENTIFIER,
+  ERC721_INTERFACE_IDENTIFIER,
+  EXCHANGE_ADDRESS,
+  isERC1155,
+  isERC721,
+  isMoney,
+  isNFT,
+  isOpenSeaFeeAccount,
+  MANTISSA_FACTOR,
+  max,
+  min,
+  Network,
+  NftStandard,
+  SaleStrategy,
+  SeaportItemType,
+  SECONDS_PER_DAY,
+  WETH_ADDRESS,
+} from "./helper";
+import { NftMetadata } from "../generated/SeaportExchange/NftMetadata";
+import { ERC165 } from "../generated/SeaportExchange/ERC165";
+
+class Sale {
+  constructor(
+    public readonly buyer: Address,
+    public readonly seller: Address,
+    public readonly nfts: NFTs,
+    public readonly money: Money,
+    public readonly fees: Fees
+  ) {}
+}
+
+class NFTs {
+  constructor(
+    public readonly collection: Address,
+    public readonly standard: string,
+    public readonly tokenIds: Array<BigInt>,
+    public readonly amounts: Array<BigInt>
+  ) {}
+}
+
+class Money {
+  constructor(public readonly amount: BigInt) {}
+}
+
+class Fees {
+  constructor(
+    public readonly protocolRevenue: BigInt,
+    public readonly creatorRevenue: BigInt
+  ) {}
+}
+
+export function handleOrderFulfilled(event: OrderFulfilled): void {
+  let offerer = event.params.offerer;
+  let recipient = event.params.recipient;
+  let offer = event.params.offer;
+  let consideration = event.params.consideration;
+
+  // I have a mixed feeling about OrderFulfilled event
+  // On the up side, it contains literally all data we need to build up a Trade entity
+  // One the down side, it is very generic and it requires a lot of hidden knowledge to decipher its actual meaning
+  // 1. `offer` doesn't tell you explicitly that if it is a bid offer or an ask offer
```

Comment:
In the `Trade` entity currently, it doesn't know the difference between a bid or ask right? As long as theres a buyer (NFT recipient) and seller (NFT seller), we can properly create a trade.

Or is `OrderFulfilled.recipient` not necessarily the NFT recipient which is why knowing the difference between bid/ask important?

Reply:
> is OrderFulfilled.recipient not necessarily the NFT recipient which is why knowing the difference between bid/ask important?

yeah, this is what I find